### ISSUE TITLE
scatter not place inputs into every device 

### ISSUE THREAD

**Issue by likelyzhao** [pytorch/pytorch#43486]
## 🐛 Bug

scatter not place inputs into every device 

## To Reproduce

import torch

x = [[[3, 2, 3],[1, 3, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
       [[4, 4, 2], [1, 1, 1]]] 
print(torch.tensor(x).shape)  
outputs = torch.cuda.comm.scatter(torch.tensor(x), [0,1,2,3], chunk_sizes=None, dim=0, streams=None)
print(len(outputs))

## Expected behavior

len(outputs)  == 4

## Environment

PyTorch version: 1.6.0
Is debug build: False
CUDA used to build PyTorch: 10.2

OS: Ubuntu 18.04.3 LTS (x86_64)
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce RTX 2080 Ti
GPU 1: GeForce RTX 2080 Ti
GPU 2: GeForce RTX 2080 Ti
GPU 3: GeForce RTX 2080 Ti

Nvidia driver version: 440.36
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.18.2
[pip3] torch==1.6.0
[pip3] torchvision==0.5.0
[conda] Could not collect

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by ngimel** [comment]
I can't reproduce this, I'm getting len(outputs)=4.

**Comment by likelyzhao** [comment]
sorry the input is. 

x = [[[3, 2, 3],[1, 3, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
    [[3, 1, 2],[3, 2, 4]],
       [[4, 4, 2], [1, 1, 1]]]
